akka-cluster/src/main/scala/akka/cluster/Cluster.scala

```diff
@@ -95,11 +100,22 @@ class Cluster(val system: ExtendedActorSystem) extends Extension {
 
   logInfo("Starting up...")
 
+  @Deprecated @deprecated("Access via `clusterTeamInternalFailureDerector` instead", since = "2.5.4")
   val failureDetector: FailureDetectorRegistry[Address] = {
-    def createFailureDetector(): FailureDetector =
+    val createFailureDetector = () ⇒
       FailureDetectorLoader.load(settings.FailureDetectorImplementationClass, settings.FailureDetectorConfig, system)
 
-    new DefaultFailureDetectorRegistry(() ⇒ createFailureDetector())
+    new DefaultFailureDetectorRegistry(createFailureDetector)
+  }
+
+  def clusterTeamInternalFailureDerector: FailureDetectorRegistry[Address] =
+    failureDetector
+
+  val crossDcFailureDetector: FailureDetectorRegistry[Address] = {
+    val createFailureDetector = () ⇒
+      FailureDetectorLoader.load(settings.CrossDcFailureDetectorSettings.ImplementationClass, settings.CrossDcFailureDetectorSettings.multiDcConfig, system)
```

Comment:
same as above but loads from it's own config section, so they can be configured separtately